src/Services/FHIR/FhirValueSetService.php

```diff
@@ -0,0 +1,184 @@
+     *  | address-types                   | postal            | Postal                |  10 |          0 |            0 |         | NULL  |       |
+     *  | address-uses                    | billing           | Billing               |  50 |          0 |            0 |         | NULL  |       |
+     *  | address-uses                    | home              | Home                  |  10 |          0 |            0 |         | NULL  |       |
+     *  | address-uses                    | old               | Old/Incorrect         |  40 |          0 |            0 |         | NULL  |       |
+     *  | address-uses                    | temp              | Temporary             |  30 |          0 |            0 |         | NULL  |       |
+     *  | address-uses                    | work              | Work                  |  20 |          0 |            0 |         | NULL  |       |
+     *  | adjreason                       | Adm adjust        | Adm adjust            |   5 |          0 |            1 |         | NULL  |       |
+     *  | adjreason                       | After hrs calls   | After hrs calls       |  10 |          0 |            1 |         | NULL  |       |
+     *  | adjreason                       | Bad check         | Bad check             |  15 |          0 |            1 |         | NULL  |       |
+     *  | adjreason                       | Bad debt          | Bad debt              |  20 |          0 |            1 |         | NULL  |       |
+     *  | adjreason                       | Coll w/o          | Coll w/o              |  25 |          0 |            1 |         | NULL  |       |
+     *
+     */
+
+
+
+    const USCGI_PROFILE_URI = 'http://hl7.org/fhir/StructureDefinition/shareablevalueset';
+
+    public function __construct()
+    {
+        parent::__construct();
+        $this->appointmentService = new AppointmentService();
+        $this->listOptionService = new ListService();
+    }
+
+    /**
+     * Returns an array mapping FHIR Resource search parameters to OpenEMR search parameters
+     */
+    protected function loadSearchParameters()
+    {
+        return [
+        '_id' => new FhirSearchParameterDefinition('_id', SearchFieldType::TOKEN, [new ServiceField('uuid', ServiceField::TYPE_UUID)]),
+        ];
+    }
+
+    /**
+     * Retrieves all of the fhir observation resources mapped to the underlying openemr data elements.
+     * @param $fhirSearchParameters The FHIR resource search parameters
+     * @return processing result
+     */
+    public function getAll($fhirSearchParameters, $puuidBind = null): ProcessingResult
+    {
+        $fhirSearchResult = new ProcessingResult();
+        try {
+              // Start with postcalendar_categories as appointment-type
+              $calendarCategories = $this->appointmentService->getCalendarCategories($fhirSearchParameters);
```

Comment:
Did something get skipped here?  AppointmentService::getCalendarCategories does not take any parameters. I thought perhaps this is where you were filtering on the ValueSet _id but it doesn't appear so as the method has not been modified.